x-pack/legacy/plugins/code/public/components/commits/commit.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  EuiButtonIcon,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiPanel,
+  EuiText,
+  EuiTextColor,
+  EuiCopy,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { CommitInfo } from '../../../model/commit';
+import { MAIN_ROOT } from '../routes';
+import { PathTypes } from '../../common/types';
+
+const COMMIT_ID_LENGTH = 8;
+
+interface ActionProps {
+  repoUri: string;
+  commitId: string;
+}
+
+const copyLabel = i18n.translate('xpack.code.commits.copyCommitSha', {
+  defaultMessage: 'Copy the full commit ID',
+});
+const revisionLinkLabel = i18n.translate('xpack.code.commits.revisionLink', {
+  defaultMessage: 'View the project at this commit',
+});
+
+const CommitActions = ({ commitId, repoUri }: ActionProps) => {
+  const revisionUri =
+    '#' +
+    MAIN_ROOT.replace(/:pathType\(.*\)/, PathTypes.tree)
+      .replace(':resource/:org/:repo', repoUri)
+      .replace(':revision', commitId);
+
+  return (
+    <div className="commit__actions">
+      <EuiCopy textToCopy={commitId}>
+        {copy => <EuiButtonIcon aria-label={copyLabel} onClick={copy} iconType="copyClipboard" />}
+      </EuiCopy>
+      <EuiButtonEmpty
+        className="commit__diff-link"
+        href={revisionUri}
+        aria-label={revisionLinkLabel}
+      >
+        <EuiText size="s">{commitId}</EuiText>
+      </EuiButtonEmpty>
+      <EuiButtonIcon href={revisionUri} aria-label={revisionLinkLabel} iconType="editorCodeBlock" />
+    </div>
+  );
+};
+
+interface Props {
+  commit: CommitInfo;
+  date: string;
+  repoUri: string;
```

Comment:
This data was blocking me from generating the `org / repo` link from the designs: `Commit` (and `CommitGroup`, and `CommitHistory`) all receive only a `repoUri` prop as identifying information.

I saw that some components use `withRouter` to pull those pieces out of the url, while others receive a `Repository` object directly from redux state.

What is the preferred practice here? I'd like to move away from generating URIs and deriving information from a string, as we seem to be doing that differently throughout the codebase.

Reply:
`withRouter` is preferred.

Reply:
@WangQianliang FYI I ended up using `RepositoryUtils` and the `repoUri` to parse out org/repo; I did not HOC the component to either redux or the router.